### PR TITLE
chore: update issue templates

### LINKED ISSUE #1571: chore: Update GitHub Isssue templates
Currently we have created two simple Issue templates

https://github.com/apache/apisix-ingress-controller/tree/master/.github/ISSUE_TEMPLATE

But they are relatively simple, and many times users need to create blank Issues to meet their needs.

I think we can do better by adding some other templates

for example:

* New feature requirements
* Documentation
* Performance issues
* question

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

**Don't merge**

Fixes #1571 

Updates issue templates to add bug report, feature request, and documentation templates.

Waiting to add a template to report performance issues.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@tao12345666333 Added a template for reporting performance issues as well. Please take a look when you are available.

File: .github/ISSUE_TEMPLATE/improve_docs.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        - Use English to communicate.
+        - Search the [open issues](https://github.com/apache/apisix-ingress-controller/issues) to avoid duplicating the issue.
+
+  - type: textarea
+    id: current-state
+    attributes:
+      label: Current State
+      description: Describe the current state of the documentation.
+      placeholder: |
+        The documentation for the API in this page (url) is missing ...
```
Comment (tao12345666333) [inline .github/ISSUE_TEMPLATE/improve_docs.yml:22]: For the function?

File: .github/ISSUE_TEMPLATE/improve_docs.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: desired-state
+    attributes:
+      label: Desired State
+      description: Describe the desired state the documentation should be in.
+      placeholder: |
+        There should be line mentioning how the API behaves when ...
```
Comment (tao12345666333) [inline .github/ISSUE_TEMPLATE/improve_docs.yml:31]: Ditto

File: .github/ISSUE_TEMPLATE/improve_docs.yml (hunk 1 above)
Comment (navendu-pottekkat) [reply .github/ISSUE_TEMPLATE/improve_docs.yml:22]: Do you mean to replace "documentation for the API" with "documentation for the function"? It is not clear to me.

Comment (tao12345666333) [reply .github/ISSUE_TEMPLATE/improve_docs.yml:22]: But in the Ingress controller project there is actually no API, only specific functions

Comment (tao12345666333) [reply .github/ISSUE_TEMPLATE/improve_docs.yml:22]: @navendu-pottekkat if you have time, please update this, thanks

Comment (navendu-pottekkat) [reply .github/ISSUE_TEMPLATE/improve_docs.yml]: Updated.

File: .github/ISSUE_TEMPLATE/improve_docs.yml (hunk 2 above)
Comment (navendu-pottekkat) [reply .github/ISSUE_TEMPLATE/improve_docs.yml]: Updated.

**Review by tao12345666333 — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**